standard/StandardProject.cmake

```diff
@@ -18,23 +18,33 @@ include(CMakeParseArguments) # Backwards compatibility
 
 # This must be a macro since project defines scope-local variables
 # that we generally rely on being in the root context.
-macro(standard_project project_name)
```

Comment:
Yeah - any reference to standard_project will have to be changed, and any explicit references to preinit and postinit should be too - I found the source of one of the comments in our toolchain file about getting parsed twice.
It turns out that if you don't have a function call to 'project' in your root CMakeLists.txt, cmake will insert a generic call for you at the very beginning of the file, which will mess up some things. Thus, our only option to have this stuff done automatically is to override the project function with our own.